### PR TITLE
feat: Add a util.Last to get last item in array

### PR DESCRIPTION
**Pull request by garrensmith**
## Describe your changes
Instead of doing lists[len-1] have a function that does this for us. Makes the code a little more readable.

## How best to test these changes
Existing tests should all pass

### CODE REVIEW

File: util/util.go [hunk 1]
```diff
@@ -65,3 +65,7 @@ func JSONToMap(data []byte) (map[string]any, error) {
 
 	return decoded, nil
 }
+
+func Last[T any](list []T) T {
```
Comment (himank) [inline util/util.go:69]: Return `null` if the list is empty.

**Review by efirs — APPROVED** [review]

File: util/util.go [hunk 2]
```diff
@@ -65,3 +65,11 @@ func JSONToMap(data []byte) (map[string]any, error) {
...
+	if len(list) == 0 {
+		var zeroVal T
+		return zeroVal
```
Comment (himank) [inline util/util.go:72]: the problem with this approach is it would return `0` for example for an integer null array which is a valid value.

Comment (himank) [reply util/util.go:72]: Or we can do this,
```
func Last[T any](list []T) (T, bool) {
	if len(list) == 0 {
		var zeroVal T
		return zeroVal, false
	}
	return list[len(list)-1], true
}
```

**Comment by garrensmith** [comment]
Closing this. I don't like the zeroval return value and I don't want to have to two return values. I'll see if I can come up with a better solution

**Pull request closed without merging**